static/app/components/confirm.stories.tsx

```diff
@@ -0,0 +1,186 @@
+    return (
+      <Fragment>
+        <p>
+          There is also a prop called <JSXProperty name="bypass" value={Boolean} />. This
+          can help to skip the Confirm dialog, for example if not enough items are
+          selected in a bulk-change operation, and directly run the{' '}
+          <JSXProperty name="onConfirm" value={Function} /> callback.
+        </p>
+        <SideBySide>
+          <Button
+            onClick={() =>
+              openConfirmModal({
+                bypass: false,
+                onRender: () => setCallbacks(prev => prev.concat('onRender')),
+                onConfirming: () => setCallbacks(prev => prev.concat('onConfirming')),
+                onCancel: () => setCallbacks(prev => prev.concat('onCancel')),
+                onConfirm: () => setCallbacks(prev => prev.concat('onConfirm')),
+                onClose: () => setCallbacks(prev => prev.concat('onClose')),
+              })
+            }
+          >
+            With callbacks (bypass = false)
+          </Button>
+          <Button
+            onClick={() =>
+              openConfirmModal({
+                bypass: true,
+                onRender: () => setCallbacks(prev => prev.concat('onRender')),
+                onConfirming: () => setCallbacks(prev => prev.concat('onConfirming')),
+                onCancel: () => setCallbacks(prev => prev.concat('onCancel')),
+                onConfirm: () => setCallbacks(prev => prev.concat('onConfirm')),
+                onClose: () => setCallbacks(prev => prev.concat('onClose')),
+              })
+            }
+          >
+            With callbacks (bypass = true)
+          </Button>
+        </SideBySide>
+        <p>
+          <label>
+            Callback debugger:
+            <br />
+            <textarea rows={4} value={callbacks.join('\n')} />
+            <br />
+            <button onClick={() => setCallbacks([])}>Reset</button>
+          </label>
+        </p>
+      </Fragment>
+    );
+  });
+
+  story('<Confirm> child render func', () => (
+    <Fragment>
+      <p>
+        Here's an example where <JSXProperty name="children" value={Function} /> is a
+        render function:
+      </p>
+      <Confirm>{({open}) => <Button onClick={open}>Open the modal</Button>}</Confirm>
+    </Fragment>
+  ));
+
+  story('<Confirm> specific props', () => {
+    const [clicks, setClicks] = useState(0);
+
+    return (
+      <Fragment>
+        <p>
+          We can see how <JSXProperty name="disabled" value={Boolean} /> and
+          <JSXProperty name="stopPropagation" value={Boolean} /> work in combination.
+        </p>
+        <p>Button clicks: {clicks} </p>
+        <Matrix
+          propMatrix={{
+            disabled: [false, true],
+            stopPropagation: [false, true],
+          }}
+          render={props => (
+            <Button onClick={() => setClicks(prev => prev + 1)}>
+              <Confirm {...props}>
+                <Green>Open modal</Green>
```

Comment:
nit: could we make the `<Green>` bit span the entire button? I noticed that if you click outside the green (on the little white bits on either side) when `stopPropagation = true`,` it'll still up the button click count

Reply:
Ya, that was part of the demo. I can do something to make it more explicit.